presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -538,6 +540,26 @@ protected Scope visitCreateView(CreateView node, Optional<Scope> scope)
             return createAndAssignScope(node, scope);
         }
 
+        @Override
+        protected Scope visitCreateFunction(CreateFunction node, Optional<Scope> scope)
+        {
+            analysis.setUpdateType("CREATE FUNCTION");
+
+            Type returnType = metadata.getType(parseTypeSignature(node.getReturnType()));
+            List<Field> fields = node.getParameters().stream()
+                    .map(parameter -> Field.newUnqualified(parameter.getName().getValue(), metadata.getType(parseTypeSignature(parameter.getType()))))
+                    .collect(toImmutableList());
+            Scope functionScope = Scope.builder()
+                    .withRelationType(RelationId.anonymous(), new RelationType(fields))
+                    .build();
+            Type bodyType = analyzeExpression(node.getBody(), functionScope).getExpressionTypes().get(NodeRef.of(node.getBody()));
+            metadata.getTypeManager().canCoerce(bodyType, returnType);
```

Comment:
I'm not sure whether there's implications at run time if the return type of body is not exactly the same as the given `returnType`. Is there valid reasons that we want to allow coercion? Why not just require the return type to be exact?

Reply:
This is to add more flexibility to the SQL functions Presto support. I can change it to be the exact match, and properly test for the coerced case before rolling out.

Reply:
Let's do exact match for now. It's easier to relax constraints than add them in the future.

Reply:
You didn't use the return value btw.

Reply:
Done